### DISCUSSION TITLE
Set custom storage path for a neutralinojs app

### DISCUSSION THREAD

**Discussion by pathange-s** [neutralinojs/neutralinojs#881] [Ideas]
There are a few requests to have a way to customize the storage path and the directory name.
Here's [one](https://github.com/neutralinojs/neutralino.js/issues/43). 

I have been looking for a workaround for this.
One way to resolve this could be by the below method:

- Currently, the `.storage` directory is created only when the user uses the `setData` function in the beginning.
- Remove `.storage` creation code from `setData` function in storage.cpp.
- Create a separate function to create `.storage` directory with a custom path and name.
- User defines storage directory name and path in `neutralino.config.json`. Or by passing flags while creating the app. By default, it would be 
```
...
"dirName": ".storage",
"dirPath": "appPath/.storage"
...
```
- User can modify the storage path by changing it in the config.json file and creating a copy of previous storage to the new path (in case he wants to restore the data)
- Use the user-defined storage path to `set` and `get` data from `storageDirectory`

**Comment by owlysk** [comment]
This will be very helpful for me
